src/components/drs/Drs.js

```diff
@@ -3,72 +3,291 @@ import axios from 'axios';
 import React from 'react';
 import {
   Switch,
-  Route
+  Route,
+  Redirect
 } from "react-router-dom";
 import { Typography } from '@material-ui/core';
+import { format } from 'date-fns';
 import DrsIndex from './pages/DrsIndex';
 import DrsShow from './pages/DrsShow';
+import NewDrs from './pages/NewDrs';
 
-const cancelToken = axios.CancelToken;
-const drsCancelToken = cancelToken.source();
+let newDate = new Date();
+newDate.setSeconds(0, 0);
+let year = newDate.getUTCFullYear();
+let month = newDate.getUTCMonth();
+let date = newDate.getUTCDate();
+let hours = newDate.getUTCHours();
+let minutes = newDate.getUTCMinutes();
+let seconds = newDate.getUTCSeconds();
 
 class Drs extends React.Component {
   constructor(props) {
     super(props);
-    this.updateActiveDrsObject = this.updateActiveDrsObject.bind(this);
+    this.getDrsObjectsList = this.getDrsObjectsList.bind(this);
     this.handleError = this.handleError.bind(this);
+    this.updateSubmitNewDrsRedirect = this.updateSubmitNewDrsRedirect.bind(this);
     this.state = {
-      activeDrsObject: null,
+      activeDrsObject: {
```

Comment:
this code of creating a "blank" DrsObject is written both here and in`drs/pages/NewDrs.js` almost exactly the same. Do both components need to run it? If so, define a single function and pass it to wherever needs to run it. It is essentially a `resetActiveDrsObject` function